src/App.js

```diff
@@ -1,13 +1,83 @@
 import React from 'react';
+import "./styles.css";
+import TodoForm from './components/TodoComponents/TodoForm';
+import TodoList from './components/TodoComponents/TodoList';
+import Header from './components/TodoComponents/HeaderComponent';
+
 
 class App extends React.Component {
   // you will need a place to store your state in this component.
   // design `App` to be the parent component of your application.
   // this component is going to take care of state, and any change handlers you need to work with your state
+
+  constructor() {
+    super();
+    this.state = {
+      todoItems: [],
+      inputText: '',
+      todo: '',
+    }
+  }
+
+  addNewItem = e => {
+    e.preventDefault();
+    if (this.state.inputText === '') {
+      this.setState({
+        todoItem: [...this.state.todoItems]
+      })
+    } else {
+      this.setState({
+        todoItems: [...this.state.todoItems, 
+          {todo: this.state.inputText,
+          id: Date.now(),
+          completed: false}],
+        inputText: ''
```

Comment:
Your addNewItem function looks good here, and you're setting each of your todoItems objects with the correct key:value pairs :heavy_check_mark: